Neos.ContentRepository.Core/Classes/Infrastructure/DbalSchemaFactory.php

```diff
@@ -0,0 +1,134 @@
+<?php
+namespace Neos\ContentRepository\Core\Infrastructure;
+
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Types;
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\NodeType\NodeTypeName;
+use Neos\ContentRepository\Core\SharedModel\Node\NodeAggregateId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\ContentStreamId;
+
+/**
+ * Provide doctrine DBAL column schema definitions for common types in the content repository to
+ * produce consistent columns across projections.
+ *
+ * @internal Because we might need to check for platform later on and generally change the input and output format of functions within.
+ */
+final class DbalSchemaFactory
+{
+    /**
+     * The NodeAggregateId is limited to 64 ascii characters and therefore we should do the same in the database.
+     * @see NodeAggregateId
+     */
+    public static function addColumnForNodeAggregateId(Table $table, string $columnName, bool $notNull): Table
+    {
+        $table->addColumn($columnName, Types::STRING)
+                ->setLength(64)
+                ->setNotnull($notNull)
+                ->setCustomSchemaOption('charset', 'ascii')
+                ->setCustomSchemaOption('collation', 'ascii_general_ci');
+
+        return $table;
+    }
```

Comment:
What about turning this into a simple:
```suggestion
    public static function columnForNodeAggregateId(string $name): Column
    {
        return (new Column($name, Type::getType(Types::STRING))
                ->setLength(64)
                ->setCustomSchemaOption('charset', 'ascii')
                ->setCustomSchemaOption('collation', 'ascii_general_ci');
    }
```
Unfortunately `Table->_addColumn()` is protected, but this could be used in the constructor..

Reply:
Yep I was also looking at _addColumn() and also would prefer that syntax, settled on what I have now though because then we start mixing between this syntax and addColumn (or write out everything in the constructor, which seemed a bit messy as well?

I can try it, but is it worth the effort right now?

Reply:
Maybe it's not really worth it, but I fear that we'll have to add more parameters for upcoming edge cases..
So, just to see how this would feel like.
Instead of
```php
$table = $schema->createTable($this->tableNamePrefix . '_node');
$table = DbalSchemaFactory::addColumnForNodeAnchorPoint($table, 'relationanchorpoint');
$table = DbalSchemaFactory::addColumnForNodeAggregateId($table, 'nodeaggregateid', false);
// ...
```
it'd be
```php
$table = new Table($this->tableNamePrefix . '_node', [
    DbalSchemaFactory::columnForNodeAnchorPoint('relationanchorpoint'),
    DbalSchemaFactory::columnForNodeAggregateId('nodeaggregateid')->setNotnull(false),
    // ...
]);
```
If you agree to that, I could take care.